Detectors/ZDC/workflow/src/ZDCDataReaderDPLSpec.cxx

```diff
@@ -78,13 +78,24 @@ void ZDCDataReaderDPLSpec::run(ProcessingContext& pc)
     mRawReader.setVerifyTrigger(mVerifyTrigger);
     LOG(info) << "Check of trigger condition during conversion is " << (mVerifyTrigger ? "ON" : "OFF");
   }
+
   uint64_t count = 0;
   for (auto it = parser.begin(), end = parser.end(); it != end; ++it) {
-    //Proccessing each page
-    count++;
+    // Proccessing each page
     auto rdhPtr = it.get_if<o2::header::RAWDataHeader>();
-    gsl::span<const uint8_t> payload(it.data(), it.size());
-    mRawReader.processBinaryData(payload, rdhPtr->linkID);
+    if (rdhPtr == nullptr) {
+      LOG(alarm) << "ZDCDataReaderDPLSpec::run - Missing RAWDataHeader on page " << count;
+    } else {
+      if (it.data() == nullptr) {
+        LOG(alarm) << "ZDCDataReaderDPLSpec::run - Null payload pointer on page " << count;
+      } else if (it.size() == 0) {
+        LOG(alarm) << "ZDCDataReaderDPLSpec::run - No payload on page " << count;
+      } else {
+        gsl::span<const uint8_t> payload(it.data(), it.size());
+        mRawReader.processBinaryData(payload, rdhPtr->linkID);
+      }
+    }
+    count++;
```

Comment:
If raw data is corrupted, it will be likely like that for the whole TF, I would simply abandon the TF sending empty output.
Producing alarm on empty payload will overflow the infologger in case the problem persists.
Please add a counter for the alarm messages printed and stop printing them after some count (not very high, e.g. 5).